superset/security/manager.py

```diff
@@ -554,17 +554,49 @@ def can_access_chart(self, chart: "Slice") -> bool:
     def get_dashboard_access_error_object(  # pylint: disable=invalid-name
         self,
         dashboard: "Dashboard",  # pylint: disable=unused-argument
+        missing_dashboard_roles: Optional[list] = None,
+        required_external_groups: Optional[list] = None,
     ) -> SupersetError:
         """
         Return the error object for the denied Superset dashboard.
 
         :param dashboard: The denied Superset dashboard
+        :param missing_dashboard_roles: The roles that the user is missing
+        :param required_external_groups: The external groups that the user is missing
         :returns: The error object
         """
 
+        message = "You don't have access to this dashboard. "
+
+        if self.is_guest_user() and not self.has_guest_access(dashboard):
+            message = (
+                "You don't have access to this dashboard because you are not "
```

Comment:
Guess this error message might not be accurate? self.is_guest_user is true, so they are a guest user?